x-pack/plugins/cases/public/components/all_cases/use_cases_columns.tsx

```diff
@@ -322,57 +286,93 @@ export const useCasesColumns = ({
-    sortable: true,
-    render: (severity: CaseUI['severity']) => {
-      if (severity != null) {
-        const severityData = severities[severity ?? CaseSeverity.LOW];
-        return (
-          <EuiHealth
-            data-test-subj={`case-table-column-severity-${severity}`}
-            color={severityData.color}
-          >
-            {severityData.label}
-          </EuiHealth>
-        );
-      }
-      return getEmptyTagValue();
-    },
-    width: '90px',
-  });
+      },
+      severity: {
+        field: casesColumnsConfig.severity.field,
+        name: casesColumnsConfig.severity.name,
+        sortable: true,
+        render: (severity: CaseUI['severity']) => {
+          if (severity != null) {
+            const severityData = severities[severity ?? CaseSeverity.LOW];
+            return (
+              <EuiHealth
+                data-test-subj={`case-table-column-severity-${severity}`}
+                color={severityData.color}
+              >
+                {severityData.label}
+              </EuiHealth>
+            );
+          }
+          return getEmptyTagValue();
+        },
+        width: '90px',
+      },
+      assignCaseAction: {
+        // no field
+        align: RIGHT_ALIGNMENT,
+        render: (theCase: CaseUI) => {
+          if (theCase.id != null) {
+            return (
+              <EuiButton
+                data-test-subj={`cases-table-row-select-${theCase.id}`}
+                onClick={() => {
+                  assignCaseAction(theCase);
+                }}
+                size="s"
+              >
+                {i18n.SELECT}
+              </EuiButton>
+            );
+          }
+          return getEmptyTagValue();
+        },
+      },
+    }),
+    [assignCaseAction, casesColumnsConfig, connectors, isSelectorView, userProfiles]
+  );
 
-  if (isSelectorView) {
-    columns.push({
-      align: RIGHT_ALIGNMENT,
-      render: (theCase: CaseUI) => {
-        if (theCase.id != null) {
-          return (
-            <EuiButton
-              data-test-subj={`cases-table-row-select-${theCase.id}`}
-              onClick={() => {
-                assignCaseAction(theCase);
-              }}
-              size="s"
-            >
-              {i18n.SELECT}
-            </EuiButton>
+  // we need to extend the columnsDict with the columns of
+  // the customFields
+  customFields.forEach(({ key, type, label }) => {
+    if (type in customFieldsBuilderMap) {
+      const columnDefinition = customFieldsBuilderMap[type]().getEuiTableColumn({ label });
+
+      columnsDict[key] = {
+        ...columnDefinition,
+        render: (theCase: CaseUI) => {
+          const customField = theCase.customFields.find(
+            (element) => element.key === key && element.value !== null
           );
-        }
-        return getEmptyTagValue();
-      },
-    });
-  }
 
-  if (!isSelectorView && actions) {
+          if (!customField) {
+            return getEmptyTagValue();
```

Comment:
Isn't this function name confusing? Consider renaming to something like getEmptyCellValue